src/P3Scheme.jl

```diff
@@ -667,4 +767,464 @@ function D_m(p3::PSP3, q::FT, N::FT, ρ_r::FT, F_r::FT) where {FT}
+    end
+
+    return v
+end
+
+function get_ice_bound(p3, λ::FT, tolerance::FT) where {FT}
+    ice_problem(x) =
+        tolerance - Γ(1 + DSD_μ(p3, λ), FT(exp(x)) * λ) / Γ(1 + DSD_μ(p3, λ))
+    guess = log(19 / 6 * (DSD_μ(p3, λ) - 1) + 39) - log(λ)
+    log_ice_x =
+        RS.find_zero(
+            ice_problem,
+            RS.SecantMethod(guess - 1, guess),
+            RS.CompactSolution(),
+            RS.RelativeSolutionTolerance(eps(FT)),
+            5,
+        ).root
+
+    return exp(log_ice_x)
+end
+
+"""
+    integration_bounds(p3, tolerance, λ, N_0, Nᵢ, qᵣ, Nᵣ)
+
+ - p3 - a struct containing P3 Scheme parameters 
+ - tolerance - tolerance to which distributions need to be evaluated to
+ - λ - shape parameter of ice distribution
+ - N_0 - intercept size distribution of ice 
+ - Nᵢ - number mixing ratio of ice
+ - q - mass mizing ratio of colliding species
+ - N - number mixing ratio of colliding species
+
+ Returns the bounds over which to integrate rain, cloud, and ice distributions to ensure 
+ coverage or more than (1 - tolerance) of each distribution
+"""
+function integration_bounds(
+    type::String,
+    p3::PSP3,
+    tolerance::FT,
+    λ::FT,
+    N_0::FT,
+    Nᵢ::FT,
+    q::FT,
+    N::FT,
+) where {FT}
+    if type == "rain"
+        rain_λ, = get_rain_parameters(q, N)
+        colliding_x = -1 / rain_λ * log(tolerance)
+    elseif type == "cloud"
+        cloud_λ, = get_cloud_parameters(q, N)
+        cloud_problem(x) =
+            tolerance -
+            exp(-exp(x)^3 * cloud_λ^3) *
+            (1 + exp(x)^3 * cloud_λ^3 + 1 / 2 * exp(x)^6 * cloud_λ^6)
+        guess =
+            log(0.5) +
+            (log(0.00025) - log(0.5)) / (log(1e12) - log(1e2)) *
+            (log(cloud_λ^3) - log(10^2))
+        log_cloud_x =
+            RS.find_zero(
+                cloud_problem,
+                RS.NewtonsMethodAD(guess),
+                RS.CompactSolution(),
+                RS.RelativeSolutionTolerance(eps(FT)),
+                5,
+            ).root
+        colliding_x = exp(log_cloud_x)
+    end
+    #= 
+        ice_problem(x) =
+            tolerance - Γ(1 + DSD_μ(p3, λ), FT(exp(x)) * λ) / Γ(1 + DSD_μ(p3, λ))
+        guess = log(19 / 6 * (DSD_μ(p3, λ) - 1) + 39) - log(λ)
+        log_ice_x =
+            RS.find_zero(
+                ice_problem,
+                RS.SecantMethod(guess - 1, guess),
+                RS.CompactSolution(),
+                RS.RelativeSolutionTolerance(eps(FT)),
+                5,
+            ).root
+     =#
+    ice_bound = get_ice_bound(p3, λ, tolerance)
+    return (2 * colliding_x, 2 * ice_bound)
+end
+
+"""
+    a(D1, D2) 
+
+ - D1 - maximum dimension of first particle 
+ - D2 - maximum dimension of second particle 
+
+ Returns the collision kernel (assumed to be of the form π(r1 + r2)^2) for the 
+ two colliding particles
+"""
+function a(D1::FT, D2::FT) where {FT}
+    # TODO make this more accurate for non-spherical particles 
```

Comment:
Do we have any ideas what to assume for the other particle regimes?

Reply:
Personally, no I do not. I was thinking finding some D' value for the regimes where the area is not in the form of D^2 such that D'^2 = D^sigma and then using D' in this calculation but I am not convinced that this would give us anything helpful.